### ISSUE TITLE
I can not pair phone with the HC-06 [problem since Marshmallow and above]

### ISSUE THREAD

**Issue by billyjoker** [mcuhq/Android-Simple-Bluetooth-Example#1]
Hi, i'm running your code, Android and Arduino both, my problem is that i can not pair my HC-06 device. However, the system bluetooth of myh phone can discover the HC-06 .... What's wrong? I see that "Discovery started" but the Broadcast Receiver never gets the connection. 

I have no crashes, all is running right.

**Comment by bauerjj** [comment]
Is your HC-06 powered properly? Does it blink its LED before you pair to it?

**Comment by billyjoker** [comment]
Yes, it is blinking. It seems to be related with bluetooth since Android Marshmallow update:

https://stackoverflow.com/questions/33052811/since-marshmallow-update-bluetooth-discovery-using-bluetoothadapter-getdefaultad

**Comment by bauerjj** [comment]
Did any of those suggestions work on that page? I don't recall the version of android I was using when testing.

**Comment by billyjoker** [comment]
The fact is that i've tested also with Android 4.4.2 and 5.0 and your code works right, so it is able to pair the HC-06 (blinking led got the permanent led ON state, meaning paired). Since Android 6.0 you have to work as the post describes.

I am trying coding that solution, when i get some results i'll tell you.

**Comment by bauerjj** [comment]
Ok - keep us posted.

**Comment by billyjoker** [comment]
Hi @bauerjj i'm been working with all kind of solutions and i can tell you that currently i am able to get work the `discovery` with my Android Marshmallow (the main problem was to discover, and consequently the pairing). The first way i did was performing the changes as described in the post i mentioned, very rude because i had to target the dicovery of bluetooth devices to Marshmallow, and the working was bad::

**BAD WAY AS DOCUMENTATION SUGGESTS (NOT TO TRY THIS WAY**)
-  To fix it you have to enable "Location" (GPS) in the settings of the phone as well as request location permission in the app at runtime. Both need to be done for scanning to work properly.
- Need to target API>=MarshMallow
- Not shows all the devices it was suppossed to be discovered

The best post that tells the above described is [this one](https://stackoverflow.com/questions/33043582/bluetooth-low-energy-startscan-on-android-6-0-does-not-find-devices?noredirect=1&lq=1)


**SOLUTION**

But i have also good news!! Finally i could get it work by doing a simple trick in tow files:

**1.** Add location permission to AndroidManifest.xml:

`<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />`

**2.** targetSdkVersion build.gradle to 22:

```
    defaultConfig {
        applicationId "com.mcuhq.simplebluetooth"
        minSdkVersion 15
        targetSdkVersion 22
        versionCode 1
        versionName "1.0"
    }
```
And that's all!! :)

**Comment by bauerjj** [comment]
That is funny. I was working on this last night after I noticed that my new android wouldn't properly scan. I also had to add the permission to the manifest, but I added code to enforce the user to enable location. Thanks!

**Comment by bauerjj** [comment]
Fixed in 908f1caf825d4b82ae7586530f39367d4c3c0893
